Patch-Source: http://patch-tracker.debian.org/patch/series/view/bzr-fastimport/0.13.0-4/dont-use-get_ancestry.diff

Description: Don't use get_ancestry() call
 The get_ancestry() method was removed from bzr sometime in the past.
 The actual code was pulled from a similar bug in bzr-stat where get_ancestry()
 is reimplemented as a two-liner.
 .
 This is just a work around until upstream has a better implementation.
Author: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1084749
Reviewed-By: Andrew Starr-Bochicchio <asb@debian.org>  Sun, 24 Nov 2013 17:46:36 -0500

--- bzr-fastimport-0.13.0.orig/branch_updater.py
+++ bzr-fastimport-0.13.0/branch_updater.py
@@ -165,7 +165,8 @@ class BranchUpdater(object):
         # apply tags known in this branch
         my_tags = {}
         if self.tags:
-            ancestry = self.repo.get_ancestry(last_rev_id)
+            graph = self.repo.get_graph()
+            ancestry = [r for (r, ps) in graph.iter_ancestry([last_rev_id]) if ps is not None]
             for tag,rev in self.tags.items():
                 if rev in ancestry:
                     my_tags[tag] = rev
